flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunHandler.java

```diff
@@ -92,12 +92,12 @@ public CompletableFuture<JarRunResponseBody> handleRequest(
             @Nonnull final DispatcherGateway gateway)
             throws RestHandlerException {
 
-        final Configuration effectiveConfiguration = new Configuration(configuration);
+        final Configuration effectiveConfiguration = new Configuration(this.configuration);
         effectiveConfiguration.set(DeploymentOptions.ATTACHED, false);
         effectiveConfiguration.set(DeploymentOptions.TARGET, EmbeddedExecutor.NAME);
 
         final JarHandlerContext context = JarHandlerContext.fromRequest(request, jarDir, log);
-        context.applyToConfiguration(effectiveConfiguration);
+        context.applyToConfiguration(effectiveConfiguration, request);
         SavepointRestoreSettings.toConfiguration(
                 getSavepointRestoreSettings(request), effectiveConfiguration);
```

Comment:
This doesn't behave correctly.

If a savepoint setting was not set in the request then we end up setting it to the default value, overwriting what was configured on the cluster.

That's what I meant with "Why isn't this using the `effectiveConfiguration`?". The fallback setting you derive in `getSavepointRestoreSettings` (which is always used as the final setting) must in part be based also on this `effectiveConfiguration`.

Reply:
fix it. now not use the restFlinkConfig to set it first,it use effectiveConfiguration to set it